datumaro/util/__init__.py

```diff
@@ -186,24 +189,31 @@ def __exit__(self, type=None, value=None, \
             return
         self._stack.__exit__(type, value, traceback)
 
+    @classmethod
+    def current(cls) -> "Rollback":
+        return cls._thread_locals.current
+
+    @contextmanager
+    def as_current(self):
+        previous = getattr(self._thread_locals, 'current', None)
+        self._thread_locals.current = self
+        try:
+            yield
+        finally:
+            self._thread_locals.current = previous
+
+# shorthand for common cases
+def on_error_do(callback, *args, ignore_errors=False):
```

Comment:
I don't think functions that take keyword arguments of their own should take generic kwargs to forward elsewhere, because then you might provoke bugs where arguments that are meant to be forwarded are used directly by the function instead. (And FWIW, I don't think `Rollback.add` should do it either.)

Also, more practically, there aren't any current uses of `on_error` that use keyword argument forwarding, so there was no reason for me to add it.